src/libs/LocaleDigitUtils.js

```
@@ -0,0 +1,82 @@
+import invariant from 'invariant';
+import _ from 'underscore';
+
+import * as NumberFormatUtils from './NumberFormatUtils';
+
+const STANDARD_DIGITS = Object.freeze(['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '.', '-', ',']);
+
+const INDEX_DECIMAL = 10;
+const INDEX_MINUS_SIGN = 11;
+const INDEX_GROUP = 12;
+
+const getLocaleDigits = _.memoize((locale) => {
+    const localeDigits = _.clone(STANDARD_DIGITS);
+    for (let i = 0; i <= 9; i++) {
+        localeDigits[i] = NumberFormatUtils.format(locale, i);
+    }
+    _.forEach(NumberFormatUtils.formatToParts(locale, 1000000.5), (part) => {
+        switch (part.type) {
+            case 'decimal':
+                localeDigits[INDEX_DECIMAL] = part.value;
+                break;
+            case 'minusSign':
+                localeDigits[INDEX_MINUS_SIGN] = part.value;
+                break;
+            case 'group':
+                localeDigits[INDEX_GROUP] = part.value;
+                break;
+            default:
+                break;
+        }
+    });
+    return Object.freeze(localeDigits);
+});
+
+/**
+ * Checks whether the given character is a valid locale digit.
+ *
+ * @param {String} locale
+ * @param {String} char A single character
+ * @returns {Boolean}
+ */
+function isValidLocaleDigit(locale, char) {
+    return getLocaleDigits(locale).includes(char);
+}
+
+/**
+ * Gets the locale digit corresponding to a standard digit.
+ *
+ * @param {String} locale
+ * @param {String} digit Character of a single standard digit . It may be "0" ~ "9" (digits),
+ * "," (group separator), "." (decimal separator) or "-" (minus sign).
+ * @returns {String}
+ *
+ * @throws If `digit` is not a valid standard digit.
+ */
+function toLocaleDigit(locale, digit) {
+    const index = _.indexOf(STANDARD_DIGITS, digit);
+    invariant(index > -1, `"${digit}" must be in ${JSON.stringify(STANDARD_DIGITS)}`);
```

We can simply throw the error here.